assembly/src/parsers/tests.rs

```diff
@@ -692,6 +692,67 @@ fn test_ast_program_serde_control_flow() {
     assert_eq!(program, program_deserialized);
 }
 
+#[test]
+fn assert_parsing_line_unmatched_begin() {
+    let source = format!("\n\nbegin\npush.1.2\n\nadd mul");
+    let err = parse_program(&source).err().unwrap();
+    assert_eq!(err, ParsingError::unmatched_begin(&Token::new("begin", 3)));
+}
+
+#[test]
+fn assert_parsing_line_extra_param() {
+    let source = format!("begin add.1.2\nend");
+    let err = parse_program(&source).err().unwrap();
+    assert_eq!(err, ParsingError::extra_param(&Token::new("add.1.2", 1)));
+}
+
+#[test]
+fn assert_parsing_line_invalid_op() {
+    let source = "\
+    begin
+        repeat.3
+            push.1
+            push.0.1
+        end
+
+        if.true
+            and
+            loc_store.0
+        else
+            padw
+        end
+
+        while.true
+            push.5.7
+            u32checked_add
+            loc_store.1
+            push.0
+        end
+
+        repeat.3
+            push.2
+            u32overflowing_mulx
+        end
+
+    end";
```

Comment:
Should we throw in a few comments (regular comment, procedure doc comment, module comment) to make sure the line numbers are still correct?

Reply:
Done! We keep track per `str::lines`, so it should work regardless of the parsing